network.go

```diff
@@ -117,8 +117,9 @@ type NetworkInterfacePair struct {
 
 // NetworkConfig is the network configuration related to a network.
 type NetworkConfig struct {
-	NetNSPath     string
-	NumInterfaces int
+	NetNSPath                   string
+	NumInterfaces               int
+	DefaultNetInterworkingModel NetInterworkingModel
```

Comment:
Well I think you're addressing this in the wrong way here. I would prefer to see a `NetInterworkingModel` being defined through the network configuration (coming from the `configuration.toml` from the runtime), and having virtcontainers deciding about a default one in case nothing is provided.
Or we could go one step further, and say that virtcontainers needs to be provided with a valid model, and handle the default value at the runtime level. The runtime will by default choose something in case the model is not provided through the `configuration.toml`.
Did I miss something ?

Reply:
Yes, `DefaultNetInterworkingModel` is not the right name here, it should be something like `InterworkingModel`. It's not the default virtcontainers model, but the one we've been asked to use from the caller.

Then probably in `initNetworkCommon` you want to set the `config.InterworkingModel` to `DefaultNetInterworkingModel` iff the provided value is invalid.

Reply:
@sboeuf  the current changes now make sense to you?

Reply:
@jcvenegas could you ping me back on Monday about that, I'll do the review properly.